src/utils/string/index.ts

```diff
@@ -167,3 +167,24 @@ export const applyStrictRegex = (str: string) => `^${str}$`;
  * @return {string} A shortenend version of the input string.
  */
 export const shortenGithash = (str: string) => str?.substring(0, 7);
+
+/**
+ * Function that trims the middle portion of a string. ex: "Hello" -> "He...lo"
+ * @param {string} text - Text to trim
+ * @param {number} maxLength - Max length before trimming text
+ * @param {number} trailingLength - Desired length of trailing text
+ * @return {string} The original or trimmed text.
+ */
+export const trimMiddleText = (
+  text: string,
+  maxLength: number,
+  trailingLength: number
+): string => {
+  if (text.length > maxLength) {
```

Comment:
This should account for the additional ellipsis otherwise we could end up with a string as long or longer as the original string.